app/frontend/src/features/donations/DonationsBoxMixed.tsx

```diff
@@ -0,0 +1,440 @@
+      const result = await stripe.redirectToCheckout({
+        sessionId: session_id,
+      });
+      if (result.error) {
+        throw Error(result.error.message);
+      }
+    },
+    {
+      onSuccess: () => {
+        resetForm();
+      },
+    }
+  );
+
+  const onSubmit = handleSubmit((data) => {
+    initiateDonation(data);
+  });
+
+  return (
+    <>
+      <form onSubmit={onSubmit} className={classes.donationsBox}>
+        {error && <Alert severity="error">{error.message}</Alert>}
+        {success && (
+          <Alert severity="success">
+            Thank you for donating to Couchers.org! We appreciate your support!
+          </Alert>
+        )}
+        {cancelled && (
+          <Alert severity="warning">The payment was cancelled.</Alert>
+        )}
+        <Typography className={classes.marginBottom2} variant="h3">
+          {DONATIONSBOX_TITLE}
+        </Typography>
+        <Controller
+          id="recurring"
+          control={control}
+          name="recurring"
+          rules={{ required: "This field is required." }}
+          defaultValue="monthly"
+          render={({ onChange, value }) => (
+            <FormControl className={classes.formGroup}>
+              <RadioGroup
+                className={classes.donationsBoxRow}
+                aria-label="recurring"
+                name="recurring-radio"
+                onChange={(e) => onChange(e.target.value === "monthly")}
+                value={value ? "monthly" : "one-time"}
+              >
+                <FormControlLabel
+                  className={classes.buttonSecondaryRadio}
+                  value="monthly"
+                  control={<Radio />}
+                  label={DONATIONSBOX_MONTHLY}
+                />
+                <FormControlLabel
+                  className={classes.buttonSecondaryRadio}
+                  value="one-time"
+                  control={<Radio />}
+                  label={DONATIONSBOX_ONETIME}
+                />
+              </RadioGroup>
+              <FormHelperText error={!!errors?.recurring?.message}>
+                {errors?.recurring?.message ??
+                  "A monthly gift helps us plan ahead and build our operations on a stable footing."}
+              </FormHelperText>
+            </FormControl>
+          )}
+        />
+
+        <Divider className={classes.marginY2} />
+
+        <Controller
+          control={control}
+          name="amount"
+          render={({ onChange, value }) => (
+
+          <div className={classes.donationsBoxRow}>
+            <div className={classes.donationsBoxSubRow}>
+              <button
+                type="button"
+                onChange={(amount) => onChange(amount)}
+                className={classNames(classes.buttonSecondary, { [classes.buttonSecondaryActive]: watchedAmount === DONATIONSBOX_VALUES[0].amount })}
+                value={DONATIONSBOX_VALUES[0].amount}
+              >
```

Comment:
```suggestion
              <button
                type="button"
                onChange={() => onChange(DONATIONSBOX_VALUES[0].amount)}
                className={classNames(classes.buttonSecondary, { [classes.buttonSecondaryActive]: value === DONATIONSBOX_VALUES[0].amount })}
              >
```

Reply:
Done that but it doesn't seem to trigger the class change (see latest push).

Reply:
Sorry, it's my typo - should of course be `<button onClick={() => onChange(...`
Styles are applying but being overriden because the non-active styles have more specificity.

Reply:
Ok, no worries, I'll look into that :)